pkgs/development/libraries/glibc/0001-nss_module.c-try-loading-NSS-modules-from-run-nss-mo.patch

```diff
@@ -0,0 +1,51 @@
+From 65a211ee3604733dceba13062f956256a573f27c Mon Sep 17 00:00:00 2001
+From: Florian Klink <flokli@flokli.de>
+Date: Sun, 19 Sep 2021 13:26:33 +0200
+Subject: [PATCH] nss_module.c: try loading NSS modules from /run/nss-modules
+ as a fallback
+
+Previously, glibc only looked for NSS modules in ${glibc.out}/lib
+and LD_LIBRARY_PATH.
+
+LD_LIBRARY_PATH is very invasive, so we don't want to set that globally
+on NixOS. We previously worked around this by running nscd with
+LD_LIBRARY_PATH set, but nscd has some caching issues, and leaks DNS
+requests across network namespaces, so it's cleaner to have glibc look
+for NSS modules in an additional path that's provided by NixOS.
+
+On non-NixOS distributions, this shouldn't change behaviour, as the path
+doesn't exist there.
+---
+ nss/nss_module.c | 16 ++++++++++++++++
+ 1 file changed, 16 insertions(+)
+
+diff --git a/nss/nss_module.c b/nss/nss_module.c
+index 6c5f341f..44cfc2a4 100644
+--- a/nss/nss_module.c
++++ b/nss/nss_module.c
+@@ -133,6 +133,22 @@ module_load (struct nss_module *module)
+       return false;
+ 
+     handle = __libc_dlopen (shlib_name);
++
++    /* After loading from the default locations, try loading from
++       /run/nss-modules, to allow loading NixOS-provided NSS modules. */
++    if(handle == NULL)
++      {
++        const char *nix_glibc_nss_path = "/run/nss-modules/lib/";
```

Comment:
```suggestion
+        const char *nix_glibc_nss_path = "/nix/var/nix/profiles/system/sw/lib";
```

Reply:
Still better than a new path that the user needs to mount into chroot or has to have control over on Non-NixOS systems.

Reply:
As already discussed in https://github.com/NixOS/nixpkgs/pull/111194#issuecomment-924017277, let's not introduce another path here, but stay with how it's done for OpenGL drivers.

Using `/run/nss-modules` is no problem on NIxOS, and this doesn't change behaviour on non-NixOS.

We can talk about how it can be solved for non-NixOS in a followup. As written there, it's probably a bad idea to have Nix manage NSS modules in first place, but we should load those provided by the distro (as they're considered system configuration).

Reply:
I still think it is. I have to special case the opengl drivers in tools like `nix-user-chroot`.
Also you are repeating that we should use the native libs in other distro - this is actually not possible without processing ldconfig. I opened up https://github.com/NixOS/nixpkgs/pull/138965